src/Markdig/Helpers/LinkHelper.cs

```diff
@@ -247,10 +246,6 @@ public static bool TryParseAutolink(ref StringSlice text, out string link, out b
                     }
                     else if (c == '.')
                     {
-                        if (pc == '.' || pc == '-')
```

Comment:
Don't remember exactly the reasoning (no comments unfortunately), but better fix by adding `pc = c` at the end of the loop rather than removing it all.